### DISCUSSION TITLE
Generating a single-color image or tiling of the same size as another image

### DISCUSSION THREAD

**Discussion by peteroupc** [ImageMagick/ImageMagick#7423] [Help] [answered]
### ImageMagick version

7.1.1-26

### Operating system, version and so on

Fedora 40

### Description

I have come across the following problem: generate an image made up of a single color, where the image's dimensions are the same as those of the input image.  This could be done with:

    -size 200x250 xc:#123456
    
...but only if the width and height of the input are known in advance.  Unfortunately, the `-size` option requires an exact value for the width and height and doesn't take an arbitrary geometry.  If `-size` accepted a geometry I could do:

    -size 100%x100% xc:#123456
    
and get an image colored #123456 with the same dimensions as the input image.

A similar problem is generating a tiling of a second image, in which case `xc:#123456` above is replaced with `tile:pattern.png`.

Is there another way to generate a single-color image or a tiling with the same dimensions as an input image, without having to know the input image's width and height in advance?

**Comment by fmw42** [comment, marked as answer]
You can create a solid color image by cloning your input image and then colorize it.

`magick image.suffix \( +clone -fill xc:"#123456" -colorize 100 \) .... output.suffix`

**Reply by peteroupc** [reply]
Thanks for your response; the single-color problem is solved.  Do you know of a solution for a tiling of a second image, a tiling which has the same size as the input image?

**Comment by fmw42** [comment]
See tiling with MPR: at https://imagemagick.org/Usage/canvas/#tile_memory

Example:

 ```
magick tree.gif  -write mpr:tile +delete \
          granite: -fill mpr:tile  -draw 'color 0,0 reset' \
          tile_mpr_reset.gif
```

**Reply by peteroupc** [reply]
Solved.

**Comment by snibgo** [comment]
There are many methods for the single-color problem. One is:
```
magick in.png -size %wx%h xc:#123456 .... out.png
```

Note that `%w` and `%h` are the width and height of the first image in the current list, which in this example is `in.png`.
